### PR TITLE
Bug 1924701: Fix cluster destroy when byo is used with Kuryr

### PR DESCRIPTION
**Pull request by MaysaMacedo**
When using byo the cluster destroy was relying on the
primary network tag to identify the router used and remove
the extra interfaces. As the provided network is not tagged
anymore the cluster is not able to get cleaned up. This commit
fixes the issue by relying on the network used by the Servers
to identify the router used.

### CODE REVIEW

**Comment by MaysaMacedo** [comment]
/hold

Needs a bit of polishing and more testing.

**Comment by MaysaMacedo** [comment]
/hold cancel

It worked fine.

File: pkg/destroy/openstack/openstack.go [hunk 1]
```diff
@@ -499,37 +510,56 @@ func deleteCustomRouterInterfaces(opts *clientconfig.ClientOpts, filter Filter,
...
+	// find the machines Network by the master machines Ports
 	tags := filterTags(filter)
-	networkTag := infraID + "-primaryClusterNetwork"
-	listOpts := networks.ListOpts{
-		Tags:    networkTag,
-		NotTags: strings.Join(tags, ","),
+	listOpts := ports.ListOpts{
+		Tags:        strings.Join(tags, ","),
+		DeviceOwner: "compute:nova",
+		NotTags:     "cluster-api-provider-openstack",
```
Comment (dulek) [inline pkg/destroy/openstack/openstack.go:518]: This could use a comment explaining why we do it.

**Review by dulek — APPROVED** [review]
/lgtm

Minor request for a comment inline.

**Comment by iamemilio** [comment]
Clever solution
/approve

**Pull request merged**